src/Orchestration/Adapter.php

```diff
@@ -38,32 +48,47 @@ abstract public function pull(string $image): bool;
     /**
      * List Containers
      *
-     * @return bool
+     * @return Container[]
      */
-    abstract public function list(): bool;
+    abstract public function list(): array;
```

Comment:
I think we should extend this to have some filtering capabilities. For example, inside the function worker in Appwrite we use the ` --filter label=appwrite-type=function` flag to list only relevant containers. This should probably introduce a new set of filter tests.

Reply:
We would need to sit down and think how the filters would work since different orchestration tools use different filters

Reply:
@PineappleIOnic waiting on our performance check for this, if results are similar, I think this logic should be implemented here and not in Appwrite itself.

Reply:
I have ran performance tests for both methods of filtering on both current adapters and the results are very interesting:
```
API using PHP Filters: 0.014989852905273438
CLI using PHP Filters: 0.04117608070373535
API using Docker Filters: 0.0006139278411865234
CLI using Docker Filters: 0.03079700469970703
```

Reply:
Native Filters have been implemented :partying_face: